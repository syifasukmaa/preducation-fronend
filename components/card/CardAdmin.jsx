import React from 'react';
import CardAdminItem from './CardAdminItem';

export default function CardAdmin() {
  return (
    <div className="mt-12 mb-8 md:mb-10 md:mt-16 md:px-12 px-4 grid md:grid-cols-2 lg:grid-cols-3 gap-4 md:gap-5">
      <CardAdminItem
        bg={'bg-dark-blue-03'}
        total={450}
        text={'Active Users'}
      />
      <CardAdminItem
        bg={'bg-alert-green'}
        total={25}
        text={'Active Class'}
      />
      <CardAdminItem
        bg={'bg-dark-blue-05'}
        total={20}
        text={'Premium Class'}
      />
    </div>
  );
}
