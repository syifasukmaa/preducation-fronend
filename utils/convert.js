const formatToCurrency = (number) => {
  const formattedNumber = new Intl.NumberFormat('id-ID', {
    style: 'currency',
    currency: 'IDR',
    minimumFractionDigits: 0,
    maximumFractionDigits: 2,
  }).format(number)

  return formattedNumber
}

const formatToDate = (date) => {
  const dates = new Date(date)

  const options = {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
    timeZone: 'Asia/Jakarta',
  }

  const formattedDate = dates.toLocaleString('id-ID', options)

  return formattedDate
}

export { formatToCurrency, formatToDate }
