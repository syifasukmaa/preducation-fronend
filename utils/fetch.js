export const loginAdmin = async (username, password) => {
  try {
    const response = await fetch(`${process.env.API_URL}/auths/admin/login`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        username,
        password,
      }),
    })
    return response
  } catch (error) {
    console.log(error)
  }
}

export const createNewCourse = async (token, courseData) => {
  try {
    const response = await fetch(`${process.env.API_URL}/courses`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(courseData),
    })

    return response
  } catch (error) {
    console.error('Error creating a new course:', error)
    throw error
  }
}

export const updateCourse = async (token, courseId, updatedCourseData) => {
  try {
    const response = await fetch(`${process.env.API_URL}/courses/${courseId}`, {
      method: 'PATCH',
      headers: {
        Authorization: `Bearer ${token}`,
      },
      body: updatedCourseData,
    })

    return response
  } catch (error) {
    console.error('Error updating the course:', error)
    throw error
  }
}

export const deleteCourse = async (token, courseId) => {
  try {
    const response = await fetch(`${process.env.API_URL}/courses/${courseId}`, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
    })

    return response
  } catch (error) {
    console.error('Error updating the course:', error)
    throw error
  }
}

export const createNewChapter = async (token, chapterData, courseId) => {
  try {
    const response = await fetch(`${process.env.API_URL}/chapters/?courseId=${courseId}`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(chapterData),
    })

    return response
  } catch (error) {
    console.error('Error creating a new course:', error)
    throw error
  }
}

export const updateChapter = async (token, chapterData, chapterId) => {
  try {
    const response = await fetch(`${process.env.API_URL}/chapters/${chapterId}`, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(chapterData),
    })

    return response
  } catch (error) {
    console.error('Error creating a new course:', error)
    throw error
  }

};
export const deleteChapter = async (token, chapterId) => {
  try {
    const response = await fetch(`${process.env.API_URL}/chapters/${chapterId}`, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
    });

    return response;
  } catch (error) {
    console.error('Error updating the course:', error);
    throw error;
  }
};

export const createNewVideo = async (token, videoData, Id) => {
  try {
    const response = await fetch(`${process.env.API_URL}/videos/?chapterId=${Id}`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(videoData),
    })

    return response
  } catch (error) {
    console.error('Error creating a new course:', error)
    throw error
  }
}

export const updateVideo = async (token, videoData, Id) => {
  try {
    const response = await fetch(`${process.env.API_URL}/videos/${Id}`, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(videoData),
    })

    return response
  } catch (error) {
    console.error('Error creating a new course:', error)
    throw error
  }
}

export const deleteVideo = async (token, videoId) => {
  try {
    const response = await fetch(`${process.env.API_URL}/videos/${videoId}`, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
    })

    return response
  } catch (error) {
    console.error('Error updating the course:', error)
    throw error
  }
}

export const resetPassword = async (passwordResetToken, password, confirmPassword) => {
  try {
    const response = await fetch(`${process.env.API_URL}/auths/reset-password/${passwordResetToken}`, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        password,
        confirmPassword,
      }),
    })

    return response
  } catch (error) {
    console.log('Error creating a new course:', error)
    throw error
  }
}
