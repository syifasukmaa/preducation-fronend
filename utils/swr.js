import useSWR from 'swr'

export const fetcher = async (url, token) => {
  const headers = {}
  if (token) {
    headers.Authorization = `Bearer ${token}`
  }
  const res = await fetch(url, {
    headers,
  })
  const data = await res.json()
  return data
}

export const useCourse = (token, id, category, title) => {
  let titleQuery = ''

  if (title) {
    titleQuery = title
  }

  const { data, isLoading, mutate, error } = useSWR(
    token && id
      ? [`${process.env.API_URL}/courses/${id}`, token]
      : [`${process.env.API_URL}/courses/?category=${category}&title=${titleQuery}`, token],
    ([url, token]) => fetcher(url, token)
  )
  return {
    course: data?.data,
    isLoading,
    mutate,
    error,
  }
}

export const useChapter = (token, id) => {
  const { data, isLoading, mutate, error } = useSWR(
    token ? [`${process.env.API_URL}/chapters/${id}`, token] : null,
    ([url, token]) => fetcher(url, token)
  )
  return {
    chapter: data?.data,
    isLoading,
    mutate,
    error,
  }
}

export const usePayment = (token, status, username, limit, page) => {
  let statusQuery = ''
  let usernameQuery = ''
  let limitQuery = ''
  let pageQuery = ''
  if (status) {
    statusQuery = status
  }
  if (username) {
    usernameQuery = username
  }
  if (limit) {
    limitQuery = limit
  }
  if (page) {
    pageQuery = page
  }

  const { data, isLoading, mutate, error } = useSWR(
    token
      ? [
          `${process.env.API_URL}/payments/all/?status=${statusQuery}&username=${usernameQuery}&limit=${limitQuery}&page=${pageQuery}`,
          token,
        ]
      : null,
    ([url, token]) => fetcher(url, token)
  )
  return {
    payment: data?.data.payments,
    totalData: data?.data.total,
    isLoading,
    mutate,
    error,
  }
}

export const useCategory = (token, statistik) => {
  const { data, isLoading, mutate, error } = useSWR(
    token ? [`${process.env.API_URL}/categories/${statistik ? 'statistik' : ''}`, token] : null,
    ([url, token]) => fetcher(url, token)
  )
  return {
    categories: data?.data,
    isLoading,
    mutate,
    error,
  }
}
