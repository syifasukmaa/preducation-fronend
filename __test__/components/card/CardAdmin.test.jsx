import React from 'react'
import { render, screen } from '@testing-library/react'
import CardAdmin from '@/components/card/CardAdmin'
import { useSession } from 'next-auth/react'
import { useCategory } from '@/utils/swr'

jest.mock('next-auth/react', () => ({
  useSession: jest.fn(),
}))

jest.mock('@/utils/swr', () => ({
  useCategory: jest.fn(),
}))

describe('CardAdmin', () => {
  it('renders loading state', async () => {
    useSession.mockReturnValueOnce({ data: { user: { accessToken: 'testAccessToken' } } })
    useCategory.mockReturnValueOnce({
      categories: null,
      error: null,
      isLoading: true,
    })
    const { container, getByText } = render(<CardAdmin />)
    expect(container).toMatchSnapshot()
    expect(getByText('loading')).toBeInTheDocument()
  })

  it('renders error state', async () => {
    useSession.mockReturnValueOnce({ data: {} })
    useCategory.mockReturnValueOnce({ categories: null, error: new Error('Test error'), isLoading: false })

    const { container, getByText } = render(<CardAdmin />)
    expect(container).toMatchSnapshot()
    expect(getByText('error')).toBeInTheDocument()
  })

  it('renders data state', async () => {
    useSession.mockReturnValueOnce({ data: { user: { accessToken: 'testAccessToken' } } })
    useCategory.mockReturnValueOnce({
      categories: { activeUsers: 10, activeClass: 5, premiumClass: 2 },
      error: null,
      isLoading: false,
    })

    render(<CardAdmin />)

    expect(screen.getByText('Active Users')).toBeInTheDocument()
    expect(screen.getByText('Active Class')).toBeInTheDocument()
    expect(screen.getByText('Premium Class')).toBeInTheDocument()
  })
})
