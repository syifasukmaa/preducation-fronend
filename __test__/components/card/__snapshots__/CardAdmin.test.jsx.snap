// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardAdmin renders error state 1`] = `
<div>
  <p>
    error
  </p>
</div>
`;

exports[`CardAdmin renders loading state 1`] = `
<div>
  <p>
    loading
  </p>
</div>
`;
